src/lib/comms/sol-mavlink.c

```diff
@@ -633,6 +633,7 @@ sol_mavlink_parse_addr_protocol(const char *str, struct sol_str_slice *addr, int
     struct sol_vector tokens;
     struct sol_str_slice slice = sol_str_slice_from_str(str);
     const void *init;
+    long int *r = (long int*)port;
```

Comment:
Casting pointers from different sizes can lead to invalid memory access or undefined behaviors in system with different endianness.
The ideal is to use a local variable and then attribute the value to port.